File: crm/app/internal/domain/app_user.go
// Package domain contains pure CRM entities and business rules.
// No imports from sqlite, http, or any adapter package.
package domain

// AppUser represents a CRM staff account mapped to a Sapo staff record.
// Role is informational in v1 (auth deferred — LAN-trust model).
type AppUser struct {
	UserID    string // UUID (TEXT in SQLite)
	StaffID   *int64 // nullable — links to dim_staff.staff_id
	Email     string
	FullName  string
	Role      string // sales | care | manager | admin
	IsActive  bool
	CreatedAt string // UTC ISO-8601 with 'Z'
	UpdatedAt string // UTC ISO-8601 with 'Z'
}

// ValidRoles lists the accepted role values. Enforcement is deferred until auth is added.
var ValidRoles = []string{"sales", "care", "manager", "admin"}

File: crm/src/domain/entities/app_user.py
"""CRM AppUser domain entity — staff accounts mapped to Sapo staff records.

Pure dataclasses; no HTTP/DB adapter imports.
"""
from __future__ import annotations

from dataclasses import dataclass
from typing import Optional

# ---------------------------------------------------------------------------
# Role constants
# ---------------------------------------------------------------------------
ROLE_SALES = "sales"
ROLE_CARE = "care"
ROLE_MANAGER = "manager"
ROLE_ADMIN = "admin"
VALID_ROLES = [ROLE_SALES, ROLE_CARE, ROLE_MANAGER, ROLE_ADMIN]


# ---------------------------------------------------------------------------
# Entity
# ---------------------------------------------------------------------------

@dataclass
class AppUser:
    """CRM staff account mapped to a Sapo staff record.

    Role is informational in v1 (auth deferred — LAN-trust model).
    """
    user_id: str        # UUID (TEXT in SQLite)
    email: str
    full_name: str
    role: str           # sales|care|manager|admin
    is_active: bool
    created_at: str     # UTC ISO-8601 with 'Z'
    updated_at: str     # UTC ISO-8601 with 'Z'
    lark_user_id: Optional[str] = None  # Lark open_id (ou_...) from CF Access JWT custom.sub
